apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx

```diff
@@ -31,37 +31,32 @@ const LIMIT_PRICE_WARNING_THRESHOLD = 0.1 // 10%
 export const CreateOrder = () => {
   const strings = useStrings()
   const navigation = useNavigateTo()
-  const {createOrder, selectedPoolChanged, unsignedTxChanged, txPayloadChanged} = useSwap()
+  const {orderData, poolPairsChanged, unsignedTxChanged} = useSwap()
   const wallet = useSelectedWallet()
   const {track} = useMetrics()
 
   const sellTokenInfo = useTokenInfo({
     wallet,
-    tokenId: createOrder.amounts.sell.tokenId,
+    tokenId: orderData.amounts.sell.tokenId,
   })
   const buyTokenInfo = useTokenInfo({
     wallet,
-    tokenId: createOrder.amounts.buy.tokenId,
+    tokenId: orderData.amounts.buy.tokenId,
   })
   const [showLimitPriceWarning, setShowLimitPriceWarning] = useState(false)
   const {isBuyTouched, isSellTouched, poolDefaulted} = useSwapTouched()
   const {poolList} = useSwapPoolsByPair({
-    tokenA: createOrder.amounts.sell.tokenId ?? '',
-    tokenB: createOrder.amounts.buy.tokenId ?? '',
+    tokenA: orderData.amounts.sell.tokenId ?? '',
+    tokenB: orderData.amounts.buy.tokenId ?? '',
   })
 
-  const bestPool = useMemo(() => {
-    if (poolList !== undefined && poolList.length > 0) {
-      return poolList.sort((a, b) => a.price - b.price).find(() => true)
-    }
-    return undefined
-  }, [poolList])
+  useEffect(() => {
+    poolPairsChanged(poolList ?? [])
+  }, [poolPairsChanged, poolList])
 
   useEffect(() => {
-    selectedPoolChanged(bestPool)
-    poolDefaulted()
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [poolDefaulted, selectedPoolChanged, bestPool?.poolId, bestPool?.price])
+    if (orderData.selectedPoolId === orderData.bestPool?.pool.poolId) poolDefaulted()
```

Comment:
can it be moved after as before inside the 3 actions?

Reply:
There are many actions that can change either selectedPoolId or the bestPool.

poolDefaulted is an app context, not lib. I think it's better if the app reacts to the state instead of trying to find every place the app might trigger an state change.